src/ext/UtilExtension/wixext/UtilFinalizeHarvesterMutator.cs

```diff
@@ -1074,18 +1074,16 @@ private void MutateRegistryValues()
         /// </summary>
         private class NativeMethods
         {
-            private const int MaxPath = 255;
-
             /// <summary>
             /// Gets the short name for a file.
             /// </summary>
             /// <param name="fullPath">Fullpath to file on disk.</param>
             /// <returns>Short name for file.</returns>
             internal static string GetShortPathName(string fullPath)
             {
-                StringBuilder shortPath = new StringBuilder(MaxPath, MaxPath);
+                StringBuilder shortPath = new StringBuilder();
 
-                uint result = GetShortPathName(fullPath, shortPath, MaxPath);
+                uint result = GetShortPathName(fullPath, null, 0);
```

Comment:
Isn't this essentially hardcoding this method to return `String.Empty`?

Reply:
No.  From the doc:
Passing NULL for this parameter and zero for cchBuffer will always return the required buffer size for a specified lpszLongPath.

Reply:
Sorry, when I said method I meant the managed method `internal static string GetShortPathName`. It looks to me that the managed method is now going to either return `String.Empty` or throw an exception.

Reply:
That is the wrapper of native method, yes?   That wasn't my experience, unless heat was then catching the exception and simply using the long path later on.   I must admit I din't go too deep after this change got heat.exe to work...  Where are you seeing the empty string?

Reply:
Since `shortPath` is no longer getting passed to the native `GetShortPathName`, I don't see any other usages of it, so it's always going to return an empty string.

`StringBuilder shortPath = new StringBuilder();`
(no more usages of shortPath)
`return shortPath.ToString()`

Reply:
Oh.  Yes...that absolutely shouldn't work.  Sorry...  and Thank you!   I may have lost something between initial testing and getting CLA looked at, etc.  I expect I should be setting shortPath to null and passing it as the second arg.  I'll re-evaluate/resubmit.  Not sure if you prefer a separate PR or just a modification of this one.

Reply:
I updated the file.  I ended up calling getShortPathName twice though, once to get the buffer size then again with the correct buffer and buffer size.  Looking in the debugger the short path name is set before returning, though I think heat.exe works even when the shortPath is null.  I would like to have assumed the short name would be shorter than the full name, but that is not guaranteed.

Reply:
This only goes so far though.  Wix.targets fails in cleanCore when it tries to find files using FindUnderPath